apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Logging/LoggingTest.cs

```diff
@@ -296,6 +296,90 @@ public async Task Logging_NoServiceProvider()
                 Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Critical));
             });
         }
+
+        [Fact]
+        public async Task Logging_ServiceContext()
+        {
+            string testId = IdGenerator.FromGuid();
+            DateTime startTime = DateTime.UtcNow;
+
+            var builder = new WebHostBuilder().UseStartup<NoBufferWarningServiceContextLoggerTestApplication>();
+            using (var server = new TestServer(builder))
+            using (var client = server.CreateClient())
+            {
+                await client.GetAsync($"/Main/Warning/{testId}");
+            }
+
+            _fixture.AddValidator(testId, results =>
+            {
+                var jsonFields = results.Single().JsonPayload.Fields;
+                Assert.Equal("my.test.service", jsonFields["serviceContext"].StructValue.Fields["service"].StringValue);
+                Assert.Equal("v1.0.0-alpha01", jsonFields["serviceContext"].StructValue.Fields["version"].StringValue);
```

Comment:
Just as an aside: if we ever get the Struct.GetValue extension method into Google.Protobuf, this would be:

```csharp
var json = results.Single().JsonPayload;
Assert.Equal("my.test.service", json.GetValue("serviceContext/service");
Assert.Equal("v1.0.0-alpha01", json.GetValue("serviceContext/version");
```

Wouldn't that be nice? :)